src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,95 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Comment } from '../types/Comment';
+import { addComment } from '../api/posts';
+
+type Props = {
+  activePostId: number,
+  postComments: Comment[],
+  setPostComments(comment: Comment[]): void,
+};
+
+interface NewComment {
+  name: string;
+  email: string;
+  body: string;
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  activePostId,
+  postComments,
+  setPostComments,
+}) => {
+  const [newComment, setNewComment] = useState<NewComment>({
+    name: '',
+    email: '',
+    body: '',
+  });
+  const [isAdding, setIsAdding] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isNameError, setIsNameError] = useState(false);
+  const [isEmailError, setIsEmailError] = useState(false);
+  const [isTextError, setIsTextError] = useState(false);
+  const addPostComments = async (comment: Omit<Comment, 'id'>) => {
+    setIsError(false);
+    setIsAdding(true);
+
+    try {
+      const response = await addComment(comment);
+
+      setPostComments([
+        ...postComments,
+        response,
+      ]);
+
+      setIsAdding(false);
+    } catch {
+      setIsError(true);
+      setIsAdding(false);
+    }
+
+    setNewComment({
+      ...newComment,
+      body: '',
+    });
+
+    setIsAdding(false);
+  };
 
-export const NewCommentForm: React.FC = () => {
   return (
-    <form data-cy="NewCommentForm">
+    <form
+      data-cy="NewCommentForm"
+      onSubmit={(event) => {
+        event.preventDefault();
+
+        if (newComment.name === '') {
```

Comment:
```suggestion
        if (!newComment.name.trim()) {
```
trim it to prevent adding spaces only as a value
empty string is a `falsy` value

Reply:
I'd still recommend you trim the value here
If you have any questions - feel free to ask in the chat